src/main/java/gregtech/api/util/GTUtility.java

```diff
@@ -635,72 +403,91 @@ public static List<FluidStack> copyFluidList(List<FluidStack> fluidStacks) {
         return Lists.newArrayList(stacks);
     }
 
-    public static ItemStack copy(ItemStack... stacks) {
+    /**
+     * @deprecated Ambiguous naming; use either {@link #copy(ItemStack)} or {@link #copyAny(ItemStack...)}
+     */
+    @Deprecated
+    @Nonnull
+    public static ItemStack copy(@Nonnull ItemStack... stacks) {
         for (ItemStack stack : stacks)
             if (!stack.isEmpty()) return stack.copy();
         return ItemStack.EMPTY;
     }
 
-    public static ItemStack copyAmount(int amount, @Nonnull ItemStack stack) {
+    /**
+     * Copies the ItemStack.
+     *
+     * @param stack item stack for copying
+     * @return a copy of ItemStack, or {@link ItemStack#EMPTY} if the stack is empty
+     */
+    @Nonnull
+    public static ItemStack copy(@Nonnull ItemStack stack) {
+        return stack.isEmpty() ? ItemStack.EMPTY : stack.copy();
+    }
+
+    /**
+     * Copies the ItemStack with new stack size.
+     *
+     * @param stack item stack for copying
+     * @return a copy of ItemStack, or {@link ItemStack#EMPTY} if the stack is empty
+     */
+    @Nonnull
+    public static ItemStack copy(int newCount, @Nonnull ItemStack stack) {
         if (stack.isEmpty()) return ItemStack.EMPTY;
         ItemStack copy = stack.copy();
-        if (amount > 64) amount = 64;
-        else if (amount == -1) amount = 111;
-        else if (amount < 0) amount = 0;
-        copy.setCount(amount);
+        copy.setCount(newCount);
         return copy;
     }
 
-    public static ItemStack copyAmount(int amount, ItemStack... stacks) {
-        ItemStack stack = copy(stacks);
-        return copyAmount(amount, stack);
-    }
-
-    public static FluidStack copyAmount(int amount, FluidStack fluidStack) {
-        if (fluidStack == null) return null;
-        FluidStack stack = fluidStack.copy();
-        stack.amount = amount;
-        return stack;
-    }
-
-    public static <T extends Comparable<T>> IBlockState[] getAllPropertyValues(IBlockState blockState, IProperty<T> property) {
-        Collection<T> allowedValues = property.getAllowedValues();
-        IBlockState[] resultArray = new IBlockState[allowedValues.size()];
-        int index = 0;
-        for (T propertyValue : allowedValues) {
-            resultArray[index++] = blockState.withProperty(property, propertyValue);
+    /**
+     * Copies first non-empty ItemStack from stacks.
+     *
+     * @param stacks list of candidates for copying
+     * @return a copy of ItemStack, or {@link ItemStack#EMPTY} if all the candidates are empty
+     * @throws IllegalArgumentException if {@code stacks} is empty
+     */
+    @Nonnull
+    public static ItemStack copyAny(@Nonnull ItemStack... stacks) {
```

Comment:
I think the name `copyFirst` might make more sense here, and for the other method too.